external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaSourceProvider.scala

```diff
@@ -560,7 +587,15 @@ private[kafka010] object KafkaSourceProvider extends Logging {
       .build()
 
   /**
-   * Returns a unique consumer group (group.id), allowing the user to set the prefix of
+   * Returns a unique batch consumer group (group.id), allowing the user to set the prefix of
+   * the consumer group
+   */
+  private[kafka010] def batchUniqueGroupId(): String = {
+    s"spark-kafka-relation-${UUID.randomUUID}"
```

Comment:
I think group ID configuration should be available on batch side as well but not yet added even if couple of lines. Don't want to mix up several intentions in this PR. I'm going to file another jira + PR  when this one merged.